nipype/algorithms/confounds.py

```diff
@@ -945,3 +1022,69 @@ def _compute_tSTD(M, x, axis=0):
     stdM[stdM == 0] = x
     stdM[np.isnan(stdM)] = x
     return stdM
+
+
+# _cosine_drift and _full_rank copied from nipy/modalities/fmri/design_matrix
```

Comment:
Originally I did it simply to avoid adding a dependency. I ended up [augmenting](https://github.com/nipy/nipype/pull/2107/commits/aed10c8f843c14773b1ceaf4845d23f68fc83cc4) it to set a minimum order of 1.